### ISSUE TITLE
Gomobile can not build Hypnic game

### ISSUE THREAD

**Issue by r3mb00** [Noofbiz/hypnic#18]
Hi, @Noofbiz.

I am in Hypnic directory and did gomobile build but got errors:
![screenshot at 2018-11-24 10-09-37](https://user-images.githubusercontent.com/9653037/48969759-274d3080-effb-11e8-9c4a-c2735227dd97.png)

Can you advice please ?

**Comment by Noofbiz** [comment]
Yeah you actually go into the androidglue folder then run 'gomobile bind --tags=mobilebind' and it'll generate an aar file that you then put into the android/androidglue folder. Then you can open android as a project in Android studio and build it as an apk there. I will put that in the readme when I get to it lol

**Comment by r3mb00** [comment]
@Noofbiz thank you.
But why gomobile build is not works here anymore?
If I am not wrong it worked before the last time I study hypnic.
Is it something related to engo engine?

**Comment by Noofbiz** [comment]
Actually, those two errors look like you're using old versions of oto and gl. In go 11 you can no longer not initialize jni objects, you have to set them explicitly to zero, so oto had to be updated for that. Then gomobile changed the teximage2d signature, so gl had to be fixed. Running `go get -u github.com/Noofbiz/hypnic` should fix the errors.

**Comment by Noofbiz** [comment]
try going into the hypnic directory and doing `git fetch` or `git checkout master`

**Comment by r3mb00** [comment]
Hi, @Noofbiz.
I handled those errors but when came to androidglue directory and execute gomobile bind --tags=mobilebind got another errors:
![screenshot at 2018-11-30 01-39-47](https://user-images.githubusercontent.com/9653037/49272873-f6da1c00-f46a-11e8-91a3-9f4701f854f7.png)

**Comment by Noofbiz** [comment]
Do you have the alsa library installed? Try 'sudo apt install libasound2-dev'

**Comment by r3mb00** [comment]
@Noofbiz thank you for helping me.
I tried as you said with libasound and terminal said:
![screenshot at 2018-12-03 06-30-42](https://user-images.githubusercontent.com/9653037/49371363-29944680-f6ef-11e8-92f9-2ddf52ed664d.png)


Then I tried again with gomobile:

![screenshot at 2018-12-03 06-31-19](https://user-images.githubusercontent.com/9653037/49371374-34e77200-f6ef-11e8-92e2-beed681d931d.png)

:(
Any suggestions?

**Comment by Noofbiz** [comment]
Looks like you have more than one version of go installed and it's trying to build both. Try uninstalling and reinstalling go. Just delete usr/local/go and then reinstall go.

**Comment by r3mb00** [comment]
@Noofbiz I reinstalled golang from binary installation to the systems installation.
Error gone but another one shows up:
![screenshot at 2018-12-06 03-34-22](https://user-images.githubusercontent.com/9653037/49571779-1a510b00-f932-11e8-8dd0-a0321844a31f.png)

**Comment by Noofbiz** [comment]
I'm not sure. It says there's a file first.go that uses package main. Any file with the package main needs to have the tag //+build !mobilebind  followed by a blank line at the top of the file. If you want to use it for Android as well you'll need a separate file with the package hypnic and //+build mobilebind

**Comment by r3mb00** [comment]
Hi, @Noofbiz 
I almost handle all appeared errors but in the end got this:
![screenshot at 2018-12-09 03-31-23](https://user-images.githubusercontent.com/9653037/49695076-1967e580-fb8d-11e8-8fbf-5a5ab4dc391e.png)
But in the platform directory I have this:
![screenshot at 2018-12-09 03-36-09](https://user-images.githubusercontent.com/9653037/49695105-88453e80-fb8d-11e8-8b6d-491081d8c0cb.png)

**Comment by r3mb00** [comment]
Yeap, just before executing this one.

**Comment by r3mb00** [comment]
Hi, @Noofbiz 
:((Got one more error:
![screenshot at 2018-12-12 02-41-48](https://user-images.githubusercontent.com/9653037/49867783-62c16a80-fe03-11e8-9cb1-8cf5b95b7205.png)
